src/stactools/sentinel3/stac.py

```diff
@@ -26,6 +26,20 @@
 logger = logging.getLogger(__name__)
 
 
+def sen3_to_kebab(asset_key):
+    """Converts asset_key to a clean kebab case"""
+    new_asset_key = ""
+    for first, second in zip(asset_key, asset_key[1:]):
+        new_asset_key += first
+        if first.islower() and second.isupper():
+            new_asset_key += "-"
+    new_asset_key += asset_key[-1]
+    new_asset_key = new_asset_key.replace("_", "-")
+    new_asset_key = new_asset_key.lower()
```

Comment:
```suggestion
    for first, second in zip(asset_key, asset_key[1:]):
        new_asset_key += first.lower()
        if first.islower() and second.isupper():
            new_asset_key += "-"
    new_asset_key += asset_key[-1].lower()
    new_asset_key = new_asset_key.replace("_", "-")
```

One fewer passes through the string.